source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,346 @@ Sorts Builders
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+
+Combining Sort Criteria
+-----------------------
+
+To combine sort criteria, use the ``Sorts.orderBy()``
+static factory method. The ``orderBy()`` method applies additional
+sort criteria from left to right in the event of ties. 
+
+In the following code snippet, we use the ``orderBy()`` method to combine a
+descending sort on ``letter`` with an ascending sort on ``_id``.
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Sorts.orderBy;
+
+   // <MongoCollection setup code here>
+
+   Bson orderBySort = orderBy(descending("letter"), ascending("_id"));
+   collection.find().sort(orderBySort);
+
+The above code snippet returns the documents in the
+:ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`
+collection in the following order: 
+
+.. code-block:: json
+
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+
+Text Score
+----------
+
+If you are using :manual:`$text </reference/operator/query/text/>`
+queries to search your data you can sort the results of those
+queries by text score (how well your search matched the returned document).
+To sort your results by the
+:manual:`$meta </reference/operator/aggregation/meta/#exp._S_meta>`
+``textScore`` associated with a ``$text`` query, use the
+``Sorts.metaTextScore()`` static factory method to retrieve a sort specification
+for a descending sort by score (best match first).
+
+.. note::
+
+  If your MongoDB instance is running MongoDB 4.2 and earlier,
+  to use a ``Sorts.metaTextScore()``
+  expression with ``FindIterable.sort()``, you must also
+  include the same expression in a projection. This is demonstrated
+  in the code example below.
+
+The ``$text`` command performs a text search on the content indexed by the 
+:manual:`text index </core/index-text/>` of your collection. You must create
+a text index on your collection before you can run a ``$text`` query and use
+``Sorts.metaTextScore()``.
+
+.. example::
+
+   In the following code example, we show how to use the
+   ``metaTextScore()`` method to sort the results of a ``$text``
+   query on
+   :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`.
+   The following code snippet works as follows:
+
+   #. Create a text index for
+      :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`
+      on the ``food`` field.
+   #. Run a ``$text`` search for the phrase "maple donut".
+   #. Project ``metaTextScore()`` into your query results as the
+      ``score`` field. This step is optional if your MongoDB instance is
+      running MongoDB 4.4 or later. 
+   #. Sort using ``metaTextScore()`` as a sort specification.
```

Comment:
Could you elaborate on what you mean by "sort specification"? I'm not familiar with what you're referring to here.

Reply:
"Sort Specification" is the language used in the [Sort Class API documentation](https://mongodb.github.io/mongo-java-driver/4.2//apidocs/mongodb-driver-core/com/mongodb/client/model/Sorts.html) when talking about Sort Criteria. You raise a good point that having two terms in the page for "Sort Criteria" seems unhelpful to readers. I will remove usages of "sort specification" and replace them with "sort criteria". Fix: https://github.com/mongodb/docs-java/pull/59/commits/bd340e849cfaa35cbaf6c2559f2a78f8cee22a31